src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -73,10 +80,6 @@
     /** The class loader used to load Checkstyle core and custom modules. */
     private final ClassLoader moduleClassLoader;
 
-    static {
-        fillShortToFullModuleNamesMap();
```

Comment:
@Luolc See comments from previous PR.
https://github.com/checkstyle/checkstyle/pull/4152#discussion_r109811131

> hardcoded list should stay to avoid bruteforce search

We only expect changes to 3rd party search, `createObjectWithIgnoringProblems` and `getAllPossibleNames`.
Instead of doing blind searches using `Class.forName`, we want to get all modules from defined package file and see if check's name matches one of those.